code/components/jomjol_image_proc/CImageBasis.cpp

```diff
@@ -470,7 +477,12 @@ CImageBasis::CImageBasis(CImageBasis *_copyfrom)
     #endif
 
     int memsize = width * height * channels;
-    rgb_image = (unsigned char*)GET_MEMORY(memsize);
+
+    if (rgb_image != NULL) {
+        free_psram_heap(std::string(TAG) + "->rgb_image (CImageBasis1)", rgb_image);
```

Comment:
It doesn't make sense to me to free a for sure non created oject variable in a constructor. This a class therefore you have to think in objects and not in variables. Objects get created using constructor and needs to be deleted after usage on object level. This happens calling delete "object". Then Destructor gets called (in desctrutor, object variables needs to be freed then).
This comments is also valid for all other constructors in this class.

Reply:
right, I will remove them again in the constructor.